TykTechnologies/tyk-pump
# [TN-8] Healthcheck endpoint issues

## Description

Changing default healthcheck endpoint port to 8083 so it doesn't have a conflict with the gateway.
Removing localhost from ListenAndServe so it can work on Docker / k8s. If host is omitted it listen on all available interfaces instead.
Adding an error check on ListenAndServe so if the port is taken, the pump fails to initialize.


I updated the README but it needs to be updated in the documentation: https://github.com/TykTechnologies/tyk-docs/blob/0d05d16c09be9231b0bfa38cabc310b8d4eee6c1/tyk-docs/content/tyk-configuration-reference/tyk-pump-configuration/tyk-pump-configuration.md

## Related Issue




Closes #256 

## How This Has Been Tested


Locally - Docker.

## Screenshots (if appropriate)

## Types of changes


## Checklist


      fork, don't request your `master`!
      *your branch* off *our latest `master`*.

## Testing instructions

- Setup a basic Tyk Pump installation.
- Start Tyk Pump with the default configuration values.
- Send a request to `http://localhost: 8080/hello` (default health check endpoint, with default port).
- Modify `tyk_pump.conf` with the following setting: `"health_check_endpoint_port": 8083`.
- Start Tyk Pump again and send a request to `http://localhost: 8083/hello`, expecting HTTP 200.
